src/lib/store.js

```diff
@@ -1,98 +1,59 @@
-import { writePublisherConsentCookie, writeVendorConsentCookie, encodePublisherConsentData, encodeVendorConsentData } from "./cookie/cookie";
+import {
+	writeConsentCookie,
+	decodeConsentData,
+	encodeConsentData
+} from "./cookie/cookie";
 import config from './config';
 import { findLocale } from './localize';
 import log from './log';
-const arrayFrom = require('core-js/library/fn/array/from');
+import { GVL, TCModel, Vector } from '@iabtcf/core';
 
 export const SECTION_INTRO = 0;
 export const SECTION_DETAILS = 1;
 export const SECTION_PURPOSES = 0;
 export const SECTION_VENDORS = 1;
 
-/**
- * Copy a data object and make sure to replace references
- * of Set objects with new ones.
- */
-function copyData(dataObject) {
-	if (typeof dataObject !== 'object') {
-		return dataObject;
-	}
-	const copy = {...dataObject};
-	for (let key in copy) {
-		if (copy.hasOwnProperty(key) && copy[key] instanceof Set) {
-			copy[key] = new Set(copy[key]);
-		}
-	}
-	return copy;
-}
-
 export default class Store {
 	constructor({
-		cmpId = 1,
-		cmpVersion = 1,
-		cookieVersion = 1,
-		vendorConsentData,
-		globalVendorConsentData,
-		publisherConsentData,
-		vendorList,
-		customPurposeList,
-		pubVendorsList,
-		allowedVendorIds
+		cmpId,
+		cmpVersion = 2,
+		cookieVersion = 2,
+		consentString
 	} = {}) {
 		// Keep track of data that has already been persisted
-		this.persistedVendorConsentData = copyData(vendorConsentData);
-		this.persistedPublisherConsentData = copyData(publisherConsentData);
-		this.persistedGlobalVendorConsentData = copyData(globalVendorConsentData);
-
 		const consentLanguage = findLocale().substr(0, 2).toUpperCase();
-		this.vendorConsentData = Object.assign(
+		const tcModel = new TCModel();
+		tcModel.cmpId = cmpId;
+		tcModel.cmpVersion = cmpVersion;
+		tcModel.isServiceSpecific = true;
+		tcModel.supportOOB = false;
+
+		// decoding to check if string is compatible
+		this.persistedConsentString = decodeConsentData(consentString) ? consentString : '';
+		this.persistedConsentData = decodeConsentData(consentString) || {};
```

Comment:
I don't know how time and resources consuming is `decodeConsentData`. Maybe it would be good idea to decode the same consent string only once:
```suggestion
		const consentData = decodeConsentData(consentString);
		this.persistedConsentString = consentData ? consentString : '';
		this.persistedConsentData = consentData || {};
```

Reply:
done